pkg/api/bzz_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"mime"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/api"
+	"github.com/ethersphere/bee/pkg/collection/entry"
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/file/splitter"
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/jsonhttp/jsonhttptest"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/manifest/jsonmanifest"
+	smock "github.com/ethersphere/bee/pkg/storage/mock"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/tags"
+)
+
+func TestBzz(t *testing.T) {
+	var (
+		bzzDownloadResource = func(addr, path string) string { return "/bzz:/" + addr + "/" + path }
+		storer              = smock.NewStorer()
+		sp                  = splitter.NewSimpleSplitter(storer)
+		client              = newTestServer(t, testServerOptions{
+			Storer:         storer,
+			ManifestParser: jsonmanifest.NewParser(),
+			Tags:           tags.NewTags(),
+			Logger:         logging.New(ioutil.Discard, 5),
+		})
+	)
+
+	t.Run("download-file-by-path", func(t *testing.T) {
+		fileName := "sample.html"
+		filePath := "test/" + fileName
+		missingFilePath := "test/missing"
+		sampleHtml := `<!DOCTYPE html>
+		<html>
+		<body>
+	
+		<h1>My First Heading</h1>
+	
+		<p>My first paragraph.</p>
+	
+		</body>
+		</html>`
+
+		var err error
+		var fileReference swarm.Address
+		var manifestFileReference swarm.Address
+
+		// save file
+
+		fileReference, err = file.SplitWriteAll(context.Background(), sp, strings.NewReader(sampleHtml), int64(len(sampleHtml)), false)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// save manifest
+
+		jsonManifest := jsonmanifest.NewManifest()
+
+		jsonManifest.Add(filePath, jsonmanifest.JSONEntry{
+			Reference: fileReference,
+			Name:      fileName,
+			Headers: http.Header{
+				"Content-Type": {"text/html", "charset=utf-8"},
+			},
+		})
+
+		manifestFileBytes, err := jsonManifest.Serialize()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		fr, err := file.SplitWriteAll(context.Background(), sp, bytes.NewReader(manifestFileBytes), int64(len(manifestFileBytes)), false)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		m := entry.NewMetadata(fileName)
+		m.MimeType = api.ManifestContentType
+		metadataBytes, err := json.Marshal(m)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		mr, err := file.SplitWriteAll(context.Background(), sp, bytes.NewReader(metadataBytes), int64(len(metadataBytes)), false)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// now join both references (mr,fr) to create an entry and store it.
+		entrie := entry.New(fr, mr)
```

Comment:
entrie? nah

Reply:
copy/paste from existing code
it looked strange to me as well, however I opted to leave it as-is